src/test/java/org/broadinstitute/hellbender/engine/GatkToolIntegrationTest.java

```diff
@@ -144,23 +167,55 @@ public void onTraversalStart() {
+
+        final SimpleInterval chr1Interval99 = new SimpleInterval("1", 99, 110);
+        final SimpleInterval chr1Interval100 = new SimpleInterval("1", 100, 110);
+        final SimpleInterval chr1Interval101 = new SimpleInterval("1", 101, 110);
+
+        final SimpleInterval chr3Interval99 = new SimpleInterval("3", 19999, 19999);
+        final SimpleInterval chr3Interval100 = new SimpleInterval("3", 20000, 20000);
+        final SimpleInterval chr3Interval101 = new SimpleInterval("3", 20001, 20001);
+
         return new Object[][]{
-                {Arrays.asList(chr1Interval, chr2Interval), IntervalFilteringVcfWriter.Mode.ANYWHERE, VariantEmitter.INTERVALS },
-                {Arrays.asList(chr1Interval, chr2Interval), IntervalFilteringVcfWriter.Mode.OVERLAPS, List.of(VariantEmitter.INT2, VariantEmitter.INT3, VariantEmitter.INT4, VariantEmitter.INT6, VariantEmitter.INT7, VariantEmitter.INT8)},
-                {Arrays.asList(chr1Interval, chr2Interval), IntervalFilteringVcfWriter.Mode.STARTS_IN, List.of(VariantEmitter.INT3, VariantEmitter.INT4, VariantEmitter.INT7,  VariantEmitter.INT8)},
-                {Arrays.asList(chr1Interval, chr2Interval), IntervalFilteringVcfWriter.Mode.ENDS_IN, List.of(VariantEmitter.INT2, VariantEmitter.INT3, VariantEmitter.INT6, VariantEmitter.INT7)},
-                {Arrays.asList(chr1Interval, chr2Interval), IntervalFilteringVcfWriter.Mode.CONTAINED, List.of(VariantEmitter.INT3, VariantEmitter.INT7)},
-                {Arrays.asList(chr1Interval, chr2Interval), null, VariantEmitter.INTERVALS},
+                {Arrays.asList(chr1Interval, chr2Interval), new ArrayList<>(), IntervalFilteringVcfWriter.Mode.ANYWHERE, VariantEmitter.INTERVALS },
+                {Arrays.asList(chr1Interval, chr2Interval), new ArrayList<>(), IntervalFilteringVcfWriter.Mode.OVERLAPS, List.of(VariantEmitter.INT2, VariantEmitter.INT3, VariantEmitter.INT4, VariantEmitter.INT6, VariantEmitter.INT7, VariantEmitter.INT8)},
+                {Arrays.asList(chr1Interval, chr2Interval), new ArrayList<>(), IntervalFilteringVcfWriter.Mode.STARTS_IN, List.of(VariantEmitter.INT3, VariantEmitter.INT4, VariantEmitter.INT7,  VariantEmitter.INT8)},
+                {Arrays.asList(chr1Interval, chr2Interval), new ArrayList<>(), IntervalFilteringVcfWriter.Mode.ENDS_IN, List.of(VariantEmitter.INT2, VariantEmitter.INT3, VariantEmitter.INT6, VariantEmitter.INT7)},
+                {Arrays.asList(chr1Interval, chr2Interval), new ArrayList<>(), IntervalFilteringVcfWriter.Mode.CONTAINED, List.of(VariantEmitter.INT3, VariantEmitter.INT7)},
+
+                // Tests specifically aimed at documenting how the --interval-merging-rule argument works in conjunction with the interval filtering
+                {Arrays.asList(chr1IntervalLeft, chr1IntervalRight), new ArrayList<>(), IntervalFilteringVcfWriter.Mode.CONTAINED, List.of(VariantEmitter.INT2)}, // Default is to merge all
```

Comment:
You might want to use `INTERVAL_MERGING_RULE_LONG_NAME`

Reply:
done but this is about as close to as friviolous as this gets since that argument is bedrock that won't change and it makes this block more unreadable \

Reply:
It is, but I really like to be able to find all the uses of an argument in tests by clicking on find uses.